### PR TITLE
o/snapstate: disk space check with snap update

### PR DESCRIPTION
**Pull request by stolowski**
Check disk space when updating single snap (and potentially of its prerequsites).

### CODE REVIEW

**Review by zyga — APPROVED** [review]
Looks good.

Do you plan to add a spread test later?

**Comment by stolowski** [comment]
> Looks good.
> 
> Do you plan to add a spread test later?

Yes, there is an existing PR #9221 introducing spread test for all disk space checks, I'm going to add refresh in a followup after 9221 and this one land.

File: overlord/snapstate/snapstate_update_test.go [hunk 1]
```diff
@@ -5335,3 +5336,75 @@ func (s *snapmgrTestSuite) TestEmptyUpdateWithChannelChangeAndAutoAlias(c *C) {
...
+	err := s.testUpdateDiskSpaceCheck(c, featureFlag, failInstallSize)
+	diskSpaceErr := err.(*snapstate.InsufficientSpaceError)
+	c.Assert(diskSpaceErr, ErrorMatches, `insufficient space in .* to perform "refresh" change for the following snaps: some-snap`)
+	c.Check(diskSpaceErr.Path, Equals, filepath.Join(dirs.GlobalRootDir, "/var/lib/snapd"))
+	c.Check(diskSpaceErr.Snaps, DeepEquals, []string{"some-snap"})
+}
+
+func (s *snapmgrTestSuite) TestUpdateDiskCheckSkippedIfDisabled(c *C) {
+	featureFlag := false
+	failInstallSize := false
```
Comment (mvo5) [inline overlord/snapstate/snapstate_update_test.go:5400]: Should this be true?

Comment (mvo5) [reply overlord/snapstate/snapstate_update_test.go:5400]: Aha, never-mind, I was just confused by the fact that the test always fails the disk-size check and this is an additional one

Comment (mvo5) [inline overlord/snapstate/snapstate_update_test.go:5400]: (nitpick) I guess for completeness  you *could* also test featureFlag := false; failInstallSize:=true (then all cases are checked). But not really important

**Review by mvo5 — APPROVED** [review]

**Pull request merged**